### ISSUE TITLE
cabal-helper-wrapper invalid exec format

### ISSUE THREAD

**Issue by Anrock** [haskell/haskell-ide-engine#952]
tl;dr: Symptoms are similar to #658. Wasn't fixed by installing `cabal` and running `cabal update`. Turned out `cabal-helper0.9.0.0-Cabal2.2.0.1` file had 0 size and after deleting it everything started working again.

Some points:
* Looks like `cabal-helper-wrapper` uses  [deprecated](https://www.addictivetips.com/windows-tips/access-windows-7-local-settings-folder/) `Local Settings` directory instead of recommended `%localappdata%. It's not openable by normal means (i.e. File Explorer) and this may mislead users.
* `HIE` log could be improved by also logging output of cabal-helper-wrapper in case of errors.
* `cabal-helper-wrapper` could've recovered from this particular error by deleting invalid file

Long version follows

***
Excerpt from hie.log:
```
2018-11-21 01:42:50.7319647 [ThreadId 5] - <--2--{"jsonrpc":"2.0","params":{"type":1,"message":"Got error while processing diagnostics: readCreateProcess: C:\\Users\\Anrock\\Projects\\haskell-ide-engine\\.stack-work\\install\\6f1bd928\\libexec\\x86_64-windows-ghc-8.4.4\\cabal-helper-0.9.0.0\\cabal-helper-wrapper.exe \"--with-ghc=C:\\\\Users\\\\Anrock\\\\AppData\\\\Local\\\\Programs\\\\stack\\\\x86_64-windows\\\\ghc-8.4.4\\\\bin\\\\ghc.exe\" \"--with-ghc-pkg=C:\\\\Users\\\\Anrock\\\\AppData\\\\Local\\\\Programs\\\\stack\\\\x86_64-windows\\\\ghc-8.4.4\\\\bin\\\\ghc-pkg.exe\" \"--with-cabal=cabal\" \"v1-style\" \"C:\\\\Users\\\\Anrock\\\\Projects\\\\gtk-test\" \"C:\\\\Users\\\\Anrock\\\\Projects\\\\gtk-test\\\\.stack-work\\\\dist\\\\7d103d30\" \"package-db-stack\" \"flags\" \"compiler-version\" \"ghc-merged-pkg-options\" \"config-flags\" \"non-default-config-flags\" \"ghc-src-options\" \"ghc-pkg-options\" \"ghc-lang-options\" \"ghc-options\" \"source-dirs\" \"entrypoints\" \"needs-build-output\" (exit 1): failed"},"method":"window/showMessage"}
```

Running this command unescaped in same directory produces this output:
```
C:\Users\Anrock\Projects\gtk-test> C:\Users\Anrock\Projects\haskell-ide-engine\.stack-work\install\6f1bd928\libexec\x86_64-windows-ghc-8.4.4\cabal-helper-0.9.0.0\cabal-helper-wrapper.exe "--with-ghc=C:\Users\Anrock\AppData\Local\Programs\stack\x86_64-windows\ghc-8.4.4\bin\ghc.exe" "--with-ghc-pkg=C:\Users\Anrock\AppData\Local\Programs\stack\x86_64-windows\ghc-8.4.4\bin\ghc-pkg.exe" "--with-cabal=cabal" "v1-style" "C:\Users\Anrock\Projects\gtk-test" "C:\Users\Anrock\Projects\gtk-test\.stack-work\dist\7d103d30" "package-db-stack" "flags" "compiler-version" "ghc-merged-pkg-options" "config-flags" "non-default-config-flags" "ghc-src-options" "ghc-pkg-options" "ghc-lang-options" "ghc-options" "source-dirs" "entrypoints" "needs-build-output"

cabal-helper-wrapper.exe: C:\Users\Anrock\Local Settings\Cache\cabal-helper\cabal-helper0.9.0.0-Cabal2.2.0.1: createProcess: invalid argument (Exec format error)
```

I've noticed strange path entry in output: `Local Settings` and tried opening it manually and got `access denied` error.

Some googling led to this [article](https://www.addictivetips.com/windows-tips/access-windows-7-local-settings-folder/). Basically it says that `Local Settings` is some kind of deprecated directory, replaced by directory junction after Windows XP and everyone should use %localappdata% instead.
However trying to `cd` to the path was success, so junction works.

`C:\Users\Anrock\Local Settings\Cache\cabal-helper\cabal-helper0.9.0.0-Cabal2.2.0.1` had size of 0 bytes, so i've deleted it and restarted hie - everything started working.

**Comment by Anrock** [comment]
Not sure if listed points have actionable items, so just FYI and in case someone else encounters this error.
